src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +358,511 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
-    }
+        if (newBus) {
+            const existingTargets = Object.getOwnPropertyNames(vm.disks);
+            const availableTarget = getNextAvailableTarget(existingTargets, newBus);
+            setDiskParams(diskParams => ({ ...diskParams, busType: newBus, target: availableTarget }));
+            // Disk device "cdrom" and bus "virtio" are incompatible, see:
+            // https://listman.redhat.com/archives/libvir-list/2019-January/msg01104.html
+        }
+    }, [diskParams.device, vm.disks]);
 
-    onValueChanged(key, value) {
-        let stateDelta = {};
-        const { storagePools, vm } = this.state;
+    useEffect(() => {
+        if (diskParams.device === "cdrom" && diskParams.busType === "virtio") {
+            // use onValueChange instead of setState in order to perform subsequent state change logic
+            // According to https://libvirt.org/formatdomain.html#hard-drives-floppy-disks-cdroms (section about 'target'),
+            // scsi is the default option for libvirt in this case too
+            const existingTargets = Object.getOwnPropertyNames(vm.disks);
+            const availableTarget = getNextAvailableTarget(existingTargets, "scsi");
+            setDiskParams(diskParams => ({ ...diskParams, busType: "scsi", target: availableTarget }));
+        }
+    }, [diskParams.device, diskParams.busType, vm.disks]);
+
+    const currentPoolRef = useRef();
+    useEffect(() => {
+        if (currentPoolRef.current === undefined) {
+            currentPoolRef.current = diskParams.storagePool;
 
-        switch (key) {
-        case 'storagePoolName': {
-            const currentPool = storagePools.find(pool => pool.name === value && pool.connectionName === vm.connectionName);
-            const prevPool = storagePools.find(pool => pool.name === this.state.storagePoolName && pool.connectionName === vm.connectionName);
-            this.setState({ storagePoolName: value, storagePoolType: currentPool.type });
             // Reset the format only when the Format selection dropdown changes entries - otherwise just keep the old selection
             // All pool types apart from 'disk' have either 'raw' or 'qcow2' format
-            if (currentPool && prevPool && ((currentPool.type == 'disk' && prevPool.type != 'disk') || (currentPool.type != 'disk' && prevPool.type == 'disk'))) {
-                // use onValueChange instead of setState in order to perform subsequent state change logic
-                this.onValueChanged('format', getDefaultVolumeFormat(value));
+            const prevPool = currentPoolRef.current;
+            if ((diskParams.storagePool?.type == 'disk' && prevPool?.type != 'disk') || (diskParams.storagePool?.type != 'disk' && prevPool?.type == 'disk')) {
+                prevPool.current = diskParams.storagePool;
+                setDiskParams(diskParams => ({
+                    ...diskParams,
+                    format: getDefaultVolumeFormat(diskParams.storagePool?.name),
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231110-154059-d19bea72-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)